### PR TITLE
feat: dark mode update #1 (ignore)

### PR DESCRIPTION
**Pull request by corquaid**
Description
---
Updating dark mode for:
- [x] TitleBar
- [x] HelpTip
- [x] Tags
- [x] Switch component
- [x] Tari mining
- [x] Merged mining 

Motivation and Context
---
#36 

How Has This Been Tested?
---
**Screenshots**
_Initial Home_
Screenshot 2022-07-06 at 12 33 11
Screenshot 2022-07-06 at 12 41 05

_Ready to Start_
Screenshot 2022-07-06 at 12 41 48
Screenshot 2022-07-06 at 12 42 04

_Mining Running_
Screenshot 2022-07-06 at 12 43 16
Screenshot 2022-07-06 at 12 43 33

_Mining Paused_
Screenshot 2022-07-06 at 12 43 56
Screenshot 2022-07-06 at 12 44 06

_Expert View Open (tab underline is missing in DarkMode, will be in upcoming PR)_
Screenshot 2022-07-06 at 12 44 30
Screenshot 2022-07-06 at 12 44 53

### CODE REVIEW

File: applications/launchpad/gui-react/src/components/Button/index.tsx [hunk 1]
```diff
@@ -49,6 +51,7 @@ const Button = ({
   size = 'medium',
   href,
   leftIcon,
+  leftIconColor,
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/Button/index.tsx:54]: Maybe we can use just `iconColor` and it would be applied to both icons?

File: applications/launchpad/gui-react/src/components/Callout/index.tsx [hunk 2]
```diff
@@ -28,7 +30,11 @@ const Callout = ({
   }
 
   return (
-    <StyledCallout $type={type} $inverted={inverted}>
+    <StyledCallout
+      $type={type}
+      $inverted={inverted}
+      style={{ background: theme.calloutBackground }}
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/Callout/index.tsx]: `StyledCallout` itself has the CSS rule for the background. The above `style={{ background: ... }}` will overwrite this.

File: applications/launchpad/gui-react/src/components/CoinsList/styles.ts [hunk 3]
```diff
@@ -22,5 +22,6 @@ export const IconWrapper = styled.span`
     width: 24px;
     height: 24px;
   }
-  color: ${({ theme }) => theme.inverted.secondary};
+  /* color: ${({ theme }) => theme.inverted.secondary}; */
+  color: yellow;
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/CoinsList/styles.ts]: Is it temporary change? If so, we would need to add at least some `@TODO`, but ideally would be to revert this change.

File: applications/launchpad/gui-react/src/components/HelpTip/index.tsx [hunk 4]
```diff
@@ -16,27 +16,29 @@ import { HelpTipProps } from './types'
  * @prop {CSSProperties} [style] - styles to apply to main wrapper element
  * @prop {boolean} [header] - whether the help tip should be rendered with additional top/bottom margin suitable for headers
  */
-const HelpTip = ({ text, cta, onHelp, style, header }: HelpTipProps) => (
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/HelpTip/index.tsx]: I think you can revert back to `() => ()` from `() => { return () }`

File: applications/launchpad/gui-react/src/components/NodeBox/index.tsx [hunk 5]
```diff
@@ -40,6 +41,12 @@ const NodeBox = ({
   children,
   testId = 'node-box-cmp',
 }: NodeBoxProps) => {
+  const setSvgGradient =
+    tag?.content === 'Ready to go' ||
+    tag?.content === 'Ready to set' ||
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/NodeBox/index.tsx]: `NodeBox` is a generic component, it should not try to interpret the `tag.content` itself. I believe that it comes from Mining, so I would suggest to make decision about this specific styling in `MiningBox` container. For instance, there are two configs:
- `const defaultConfig = { .... } ` around line 131
- `const defaultStates = { .... } ` around line 150

You can add another optional prop for the tag color.

And btw, two notes, but if you apply above, then these won't be applicable:
a) prefix `set` in `setSvgGradient` suggest an _action_. In this case just `svgGradient` would be good enough
b) It would be better to compare against locale rather than static text. 

```
// So it should be:
tag?.content === t.common.readyToGo
// ..instead of:
tag?.content === 'Ready to go'
```

File: applications/launchpad/gui-react/src/components/TabContent/index.tsx [hunk 6]
```diff
@@ -19,7 +19,7 @@ const TabContent = ({
     <StyledTabContent>
       <TabMainText>{text}</TabMainText>
       {running && !pending ? (
-        <Tag variant='small' type='running' subText={tagSubText}>
+        <Tag variant='small' type='running' subText={tagSubText} dashboard>
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/TabContent/index.tsx]: NIT: Not sure if `dashboard` is a good name here... I would rather try to keep things in `components/*` generic as much as possible. So instead of using a _reason_ as a prop name, you can name the prop after the _effect_ that it applies. For example, if something on the dashboard should be large, then the prop should be named `large` rather than `dashboard`.

File: applications/launchpad/gui-react/src/components/TitleBar/index.tsx [hunk 7]
```diff
@@ -80,6 +81,14 @@ const TitleBar = ({
     }
   }
 
+  const getSettingsIconColor = () => {
+    if (expertView !== 'hidden') {
+      return colors.light.textSecondary
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/TitleBar/index.tsx]: Not sure if we would like to extract this directly from the `colors`. I would suggest to even name directly this style in the themes, like: `titleBarIcon` (+ one to the `inverted` section).

Also, maybe we could make small improvement here like this:

```
const settingsIconColor =  useMemo(() => {
    if (expertView !== 'hidden') {
      return colors.light.textSecondary
    } else {
      return theme.helpTipText
    }
  }, [expertView])
```

File: applications/launchpad/gui-react/src/components/TitleBar/index.tsx [hunk 8]
```diff
@@ -207,6 +216,7 @@ const TitleBar = ({
             variant='text'
             size='small'
             leftIcon={<SvgSetting width='16px' height='16px' />}
+            leftIconColor={getSettingsIconColor()}
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/TitleBar/index.tsx]: ...and then we can use here: `leftIconColor={settingsIconColor}`

File: applications/launchpad/gui-react/src/store/settings/selectors.ts [hunk 9]
```diff
@@ -17,6 +17,7 @@ export const selectServiceSettings = (state: RootState) => ({
   moneroUsername: state.mining.merged.authentication?.username || '',
   moneroPassword: state.mining.merged.authentication?.password || '',
   rootFolder: state.baseNode.rootFolder,
+  walletPassword: state.settings.serviceSettings.parole,
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/store/settings/selectors.ts]: Possibly, no longer needed. You may need to rebase first.

**Review by tomaszantas — CHANGES_REQUESTED** [review]
Found few issues

File: applications/launchpad/gui-react/src/components/Button/index.tsx (hunk 1 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/Button/index.tsx:54]: We could but I haven't seen a use-case for that so far.

File: applications/launchpad/gui-react/src/components/Callout/index.tsx (hunk 2 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/Callout/index.tsx]: Good spot, fixed.

File: applications/launchpad/gui-react/src/components/CoinsList/styles.ts (hunk 3 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/CoinsList/styles.ts]: Error, fixed.

File: applications/launchpad/gui-react/src/components/HelpTip/index.tsx (hunk 4 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/HelpTip/index.tsx]: Fixed.

File: applications/launchpad/gui-react/src/components/NodeBox/index.tsx (hunk 5 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/NodeBox/index.tsx]: Fixed (eventually!)

File: applications/launchpad/gui-react/src/components/TabContent/index.tsx (hunk 6 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/TabContent/index.tsx]: Naming improved.

File: applications/launchpad/gui-react/src/components/TitleBar/index.tsx (hunk 7 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/TitleBar/index.tsx]: Changed to `theme.textSecondary` as it already exists.

~~`useMemo` causes issues here, breaks the icon colour when `expertView !== 'hidden'. Possibly not re-rendering when required?~~

nvm, `expertView` needed to go in the `useMemo` dependencies - all fixed now.

File: applications/launchpad/gui-react/src/components/TitleBar/index.tsx (hunk 8 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/components/TitleBar/index.tsx]: Done.

File: applications/launchpad/gui-react/src/store/settings/selectors.ts (hunk 9 above)
Comment (corquaid) [reply applications/launchpad/gui-react/src/store/settings/selectors.ts]: Checked and removed.

**Comment by corquaid** [comment]
Git history messed up, PR closed.
